ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala

```diff
@@ -322,9 +322,9 @@ class ContractsService(
     \/.fromTryCatchNonFatal(LfValueCodec.apiValueToJsValue(a)).leftMap(e =>
       Error('lfValueToJsValue, e.description))
 
-  private def resolveTemplateIds(
-      xs: Set[domain.TemplateId.OptionalPkg],
-  ): (Set[domain.TemplateId.RequiredPkg], Set[domain.TemplateId.OptionalPkg]) = {
+  private[http] def resolveTemplateIds[Tid <: domain.TemplateId.OptionalPkg](
```

Comment:
just curious why polymorphic type here? There only one type that would be accepted here: `Tid <: domain.TemplateId.OptionalPkg`

Reply:
Just making the type a little more self-documenting; every element in the right set must be an element from the argument set, arbitrary constructed `TemplateId.OptionalPkg`s won't type-check.